### ISSUE TITLE
Feature: Information banners

### ISSUE THREAD

**Issue by abias** [moodle-an-hochschulen/moodle-theme_boost_union#4]
Boost Campus provides an information banner feature. It consists of a perpetual information banner and a time controlled information banner. The documenation can be found on https://github.com/moodle-an-hochschulen/moodle-theme_boost_campus#7-tab-info-banner-settings.

It is requested to include a similar functionality into Boost Union.

Before starting the implementation (or just copying the code from Boost Campus), these additional thoughts should be considered:
* Boost Campus has a hardcoded order of one time controlled banner and one perpetual banner. It might be useful to let the admin control the order of both banners. Similarly, it might be useful to let the admin create more than two banners. See https://github.com/moodle-an-hochschulen/moodle-theme_boost_campus/issues/100 for some more thoughts.
* Boost Campus has a feature to allow Moodle users to dismiss the (perpetual) information banner and allows admins to re-show the banner for all users who dismissed the banner. It might be useful to allow each user to re-show the banner as well, similarly to how users are allowed to re-show user tours.
* The information banner can be especially used to show maintainance period informations to users. It might be useful to provide a CLI script for handling the banners. See https://github.com/moodle-an-hochschulen/moodle-theme_boost_campus/issues/101 for some more thoughts.
* The information banner functionality can get quite huge and might not necessarily have to be a part of a theme. It might be worthwhile to think about outsourcing the information banner feature into a plugin of its own.
* With https://moodle.org/plugins/auth_outage and https://moodle.org/plugins/block_advnotifications, there are two existing implementations for the similar functionality. It might be worthwhile to look at the first to avoid re-inventing the wheel.

**Comment by abias** [comment]
In #54  , I pushed a PR which realizes the groundwork for this issue. It is still based on the info banners from Boost Campus and uses @lucaboesch 's backport in #22 as a start, but I went further.

These are the most important differences to the Boost Campus original:
* The admin is now able to configure 5 info banners. The info banners are not specialized anymore into perpetual and time based banners, this decision is now an option within each banner configuration. "5" is just a constant in the code, we could extend this to even more banners or make it an admin setting in the future if needed.
* Against that background, I deduplicated much of the original Boost Campus code.
* The admin is able to control the output order of the 5 info banners.
* I dropped the confirmation dialogue when dismissing a banner (as it seemed not to be very practical)
* The reset of dismissed banners has now a proper action button and confirmation dialogue and isn't just a misused checkbox admin setting.
* I used mustache templates instead of html_writer for the output of the info banners.
* The time based banner configuration got a dropdown date configuration instead of a text string configuration.
